plugins/SegmentEditor/SegmentEditor.php

```diff
@@ -71,6 +85,92 @@ public function getKnownSegmentsToArchiveForSite(&$segments, $idSite)
         $segments = array_unique($segments);
     }
 
+    public function onNoArchiveData()
+    {
+        $segmentInfo = $this->getSegmentIfIsUnprocessed();
+        if (empty($segmentInfo)) {
+            return;
+        }
+
+        list($segment, $isSegmentToPreprocess) = $segmentInfo;
+
+        // this archive has no data, the report is for a segment that gets preprocessed, and the archive for this
+        // data does not exist. this means the data will be processed later. we let the user know so they will not
+        // be confused.
+        $model = new Model();
+        $storedSegment = $model->getSegmentByDefinition($segment->getString()) ?: null;
+
+        throw new UnprocessedSegmentException($segment, $isSegmentToPreprocess, $storedSegment);
```

Comment:
Can there be a case where the segment used to exist, but not anymore, but we did archive this segment when it existed a few weeks ago but because there was really no match for this segment that time and the report therefore actually had no data, we would trigger this exception?

Reply:
Performed a test where I:

* Enabled browser archiving & loaded a preprocessed segment on a day w/ no data.
* Verified that there was an archive w/ the segment hash.
* Disabled browser archiving & loaded the same reports (w/ same day & segment).
* Saw that the notification was not displayed.
* Went to a more recent day (after dropping the archive tables) & saw that the notification was displayed.

So this shouldn't be an issue.